debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java

```diff
@@ -37,7 +37,22 @@ public class DisableAllBreakpointsAction extends AbstractRemoveAllActionDelegate
 
 	@Override
 	protected boolean isEnabled() {
-		return DebugPlugin.getDefault().getBreakpointManager().getBreakpoints().length > 0;
+		boolean allDisabled = true;
+		IBreakpointManager breakpointManager = DebugPlugin.getDefault().getBreakpointManager();
+		IBreakpoint[] breakpoints = breakpointManager.getBreakpoints();
+		for (IBreakpoint bp : breakpoints) {
+			try {
+				if (bp.isEnabled()) {
+					return true;
+				}
+			} catch (CoreException e) {
+				DebugUIPlugin.log(e);
+			}
+		}
+		if (breakpoints.length > 0 && allDisabled) {
+			return false;
+		}
+		return true;
```

Comment:
i guess it should return false even if breakpoints.lenght==0

Reply:
okay

Reply:
Is it intentional to return true here - i.e. for no breakpoints?

Reply:
Without conditions also work, directly returning false. I think thats simpler

Reply:
IBreakpointManager breakpointManager = DebugPlugin.getDefault().getBreakpointManager();
		IBreakpoint[] breakpoints = breakpointManager.getBreakpoints();
		for (IBreakpoint bp : breakpoints) {
			try {
				if (bp.isEnabled()) {
					return true;
				}
			} catch (CoreException e) {
				DebugUIPlugin.log(e);
			}
		}
		return false;

Reply:
Optimised further and tested.